### PR TITLE
Sklearn 0.20.3 update

### PR DESCRIPTION
**Pull request by oleksandr-pavlyk**
Updated scikit-learn patches to work with scikit-learn 0.20.3

Also, deselected tests are now specified in `./deselected_tests.yaml`, which is parsed and transfored into sequence of `--deselect` commands in a Python script.

@fschlimb

### CODE REVIEW

File: deselected_tests.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+  - svm/tests/test_svm.py::test_precomputed
+  - svm/tests/test_sparse.py::test_sparse
+  - svm/tests/test_sparse.py::test_svc_iris
+  - ensemble/tests/test_bagging.py::test_sparse_classification
+  # test_k_means_fit_predict is known to sporadically fail for float32 inputs in multithreaded runs
+  # See: https://github.com/IntelPython/daal4py/issues/25
+  - cluster/tests/test_k_means.py::test_k_means_fit_predict
+  # test_non_uniform_strategies fails due to differences in handling of vacuous clusters after update
+  # See https://github.com/IntelPython/daal4py/issues/69
+  - preprocessing/tests/test_discretization.py::test_nonuniform_strategies
```
Comment (fschlimb) [inline deselected_tests.yaml]: Going forward we probably need to deselect tests depending on the sklearn version.

File: daal4py/sklearn/monkeypatch/svm.py [hunk 2]
```diff
@@ -262,15 +266,95 @@ def _daal4py_fit(self, X, y_inp, kernel):
...
+                                  "from 'auto' to 'scale' in version 0.22 to "
+                                  "account better for unscaled features. Set "
+                                  "gamma explicitly to 'auto' or 'scale' to "
+                                  "avoid this warning.", FutureWarning)
+                self._gamma = 1.0 / X.shape[1]
+        elif self.gamma == 'auto':
+            self._gamma = 1.0 / X.shape[1]
+        else:
+            self._gamma = self.gamma
+"""
```
Comment (fschlimb) [inline daal4py/sklearn/monkeypatch/svm.py]: I guess this can be removed

**Comment by oleksandr-pavlyk** [comment]
@fschlimb Ready to merge?

File: .circleci/deselect_tests.py [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+        return LooseVersion(v) <= LooseVersion(cond[2:])
+    if cond.startswith("=="):
+        return LooseVersion(v) == LooseVersion(cond[2:])
+    if cond.startswith("!="):
+        return LooseVersion(v) != LooseVersion(cond[2:])
+    if cond.startswith("<"):
+        return LooseVersion(v) < LooseVersion(cond[2:])
+    if cond.startswith(">"):
+        return LooseVersion(v) > LooseVersion(cond[2:])
+    return False
```
Comment (fschlimb) [inline .circleci/deselect_tests.py]: Shouldn't the last 2 compare against cond[**_1_**:]?

I am slightly concerned about not getting any meaningful message if using a wrong syntax. But it's probably ok to not over-engineer this.

Comment (oleksandr-pavlyk) [reply .circleci/deselect_tests.py]: Yes, good catch! I'll fix it right away.

Comment (oleksandr-pavlyk) [reply .circleci/deselect_tests.py]: Fixed in e9f51e1eff2a727515f6bfb9abb7bdc0dce5b8bd. Unclear why GH does not mark the code chunk associated with the comment as outdates.

File: daal4py/sklearn/monkeypatch/svm.py (hunk 2 above)
Comment (oleksandr-pavlyk) [reply daal4py/sklearn/monkeypatch/svm.py]: removed.

File: deselected_tests.yaml (hunk 1 above)
Comment (oleksandr-pavlyk) [reply deselected_tests.yaml]: sklearn versioning in the yaml was implemented.

File: daal4py/sklearn/monkeypatch/svm.py [hunk 4]
```diff
@@ -262,15 +266,62 @@ def _daal4py_fit(self, X, y_inp, kernel):
...
     except AttributeError:
-        return np.std(X)
+        return np.var(X)
     ssc = alg.compute(X.reshape(-1,1)).sumSquaresCentered
-    return np.sqrt(ssc[0, 0] / X.size)
+    return ssc[0, 0] / X.size
+
+
+def __compute_gamma__(gamma, kernel, X, sparse, use_var=True):
+    if gamma in ('scale', 'auto_deprecated'):
```
[the remark this answers is not in the document]
Comment (oleksandr-pavlyk) [reply daal4py/sklearn/monkeypatch/svm.py]: done.

File: deselected_tests.yaml [hunk 5]
```diff
@@ -0,0 +1,14 @@
+deselected_tests:
+  # Deselecting 5 SVC related tests where duplicate samples end up being support vectors
```
[the remark this answers is not in the document]
Comment (oleksandr-pavlyk) [reply deselected_tests.yaml]: done too.

**Pull request merged**